applications/RANSApplication/tests/cpp/test_incompressible_potential_flow_conditions.cpp

```diff
@@ -0,0 +1,282 @@
+
+    ModelPart& r_model_part = CreateScalarVariableTestModelPart(
+        rModel, "Element2D3N",
+        "RansIncompressiblePotentialFlowPressureBodyForce2D2N",
+        add_variables_function, PRESSURE_POTENTIAL, 1, true, false);
+
+    // set nodal historical variables
+    RandomFillNodalHistoricalVariable(r_model_part, DENSITY, 2.0, 2.0);
+    RandomFillNodalHistoricalVariable(r_model_part, BODY_FORCE, -2.0, 2.0);
+    RandomFillNodalHistoricalVariable(r_model_part, VELOCITY_POTENTIAL, -10.0, 10.0);
+    RandomFillNodalHistoricalVariable(r_model_part, PRESSURE_POTENTIAL, -10.0, 10.0);
+
+    RandomFillContainerVariable<ModelPart::ConditionsContainerType, array_1d<double, 3>>(
+        r_model_part, NORMAL, 1.0, 10.0);
+
+    return r_model_part;
+}
+
+} // namespace
+
+KRATOS_TEST_CASE_IN_SUITE(RansIncompressiblePotentialFlowVelocityInlet2D2N_EquationIdVector,
+                          KratosRansFastSuite)
+{
+    // Setup:
+    Model model;
+    auto& r_model_part = RansIncompressiblePotentialFlowVelocityInlet2D2N_SetUp(model);
+
+    // Test:
+    RansApplicationTestUtilities::TestEquationIdVector<ModelPart::ConditionsContainerType>(r_model_part);
+}
+
+KRATOS_TEST_CASE_IN_SUITE(RansIncompressiblePotentialFlowVelocityInlet2D2N_GetDofList, KratosRansFastSuite)
+{
+    // Setup:
+    Model model;
+    auto& r_model_part = RansIncompressiblePotentialFlowVelocityInlet2D2N_SetUp(model);
+
+    // Test:
+    RansApplicationTestUtilities::TestGetDofList<ModelPart::ConditionsContainerType>(
+        r_model_part, VELOCITY_POTENTIAL);
+}
+
+KRATOS_TEST_CASE_IN_SUITE(RansIncompressiblePotentialFlowVelocityInlet2D2N_CalculateLocalSystem,
+                          KratosRansFastSuite)
+{
+    // Setup:
+    Model model;
+    auto& r_model_part = RansIncompressiblePotentialFlowVelocityInlet2D2N_SetUp(model);
+
+    // Test:
+    Matrix LHS, ref_LHS;
+    Vector RHS, ref_RHS;
+    auto& r_condition = r_model_part.Conditions().front();
+
+    // checking for non-inlet condition
+    r_condition.SetValue(RANS_IS_INLET, 0);
+    r_condition.Initialize();
+    r_condition.CalculateLocalSystem(
+        LHS, RHS, static_cast<const ProcessInfo>(r_model_part.GetProcessInfo()));
+    // setting reference values
+    ref_RHS = ZeroVector(2);
+    ref_LHS = ZeroMatrix(2, 2);
+
+    KRATOS_CHECK_VECTOR_NEAR(RHS, ref_RHS, 1e-12);
+    KRATOS_CHECK_MATRIX_NEAR(LHS, ref_LHS, 1e-12);
+
+    // checking for inlet condition
+    r_condition.SetValue(RANS_IS_INLET, 1);
+    r_condition.Initialize();
+    r_condition.CalculateLocalSystem(
+        LHS, RHS, static_cast<const ProcessInfo>(r_model_part.GetProcessInfo()));
+    // settting reference values
+    ref_RHS[0] = -7.4912949377611748e-01;
+    ref_RHS[1] = -7.4912949377611748e-01;
+
+    KRATOS_CHECK_VECTOR_NEAR(RHS, ref_RHS, 1e-12);
+    KRATOS_CHECK_MATRIX_NEAR(LHS, ref_LHS, 1e-12);
+}
+
+KRATOS_TEST_CASE_IN_SUITE(RansIncompressiblePotentialFlowVelocityInlet2D2N_CalculateLeftHandSide,
```

Comment:
```suggestion
KRATOS_TEST_CASE_IN_SUITE(RansIncompressiblePotentialFlowVelocityInlet2D2NCalculateLeftHandSide,
```

Reply:
This I would like to keep as it is because its a long name, having underscores gives little bit of readability :)